### ISSUE TITLE
openL returns handle with extra refcount

### ISSUE THREAD

**Issue by andrewgregory** [rdboisvert/Win32-LongPath#3]
``` perl
use Win32::LongPath;
use Devel::Refcount qw( refcount );
Win32::LongPath::openL(\(my $fh), '<', $0);
print refcount($fh);   # prints 2
```

Under Strawberry Perl 5.22.1.2, `openL` results in a handle with a refcount of 2.  Due to the extra ref, the handle's refcount never reaches 0 and it is never closed unless the caller explicitly closes it.  It's not entirely clear to me how the fdopen-style usage of `open` is intended to work, so this may be a bug in Perl.  However, it appears to work correctly if instead of using the ref provided to `openL` directly in the call to `open`, an intermediate handle is created and assigned to the destination ref:

``` perl
my $fh;
if (!CORE::open ($fh, "$sMode&=$nFD"))
  { return; }
$$oFH = $fh;
```

**Comment by rdboisvert** [comment]
I'm not very familiar w/Strawberry Perl but I will look into this.

**Comment by andrewgregory** [comment]
I just tested with v5.22.1.2201 of ActivePerl and was able to replicate there as well.

**Comment by andrewgregory** [comment]
I've traced this to commit Perl/perl5@7bdb4ff0943cf93297712faf504cdd425426e57f and open a bug on the perl tracker https://rt.perl.org/Ticket/Display.html?id=127692.  Hopefully this will be fixed upstream, but given the simplicity of the workaround it would be nice if this could be fixed in Win32::LongPath regardless of what upstream does.

**Comment by rdboisvert** [comment]
This appears to be an issue with Perl and not something that can be changed at the module level.

**Comment by andrewgregory** [comment]
I understand that having to work around issues in Perl is not ideal, but is the solution I presented in the initial report not acceptable?

**Comment by rdboisvert** [comment]
Sorry, I was moving quickly and overlooked your suggestion.  I will revisit this issue.

**Comment by rdboisvert** [comment]
Pull request inadvertently closed this.  Reopening for testing.

**Comment by rdboisvert** [comment]
Released v1.06 to CPAN.  Thank you for your help in fixing this.
